.github/workflows/ci.yml

```diff
@@ -56,3 +56,85 @@ jobs:
 
       - name: Test
         run: yarn test
+
+
+  test-floating-dependencies:
+    name: Test floating dependencies
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Check out a copy of the repo
+        uses: actions/checkout@v3
+
+      - name: Use Node.js ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v3
+        with:
+          cache: 'yarn'
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Install dependencies
+        run: yarn install --no-lockfile
+
+      - name: Test
+        run: yarn test:ember
+
+
+  try-scenarios:
+    name: Try scenarios
+    runs-on: ubuntu-latest
+    needs: [lint, test-addon]
+    strategy:
+      fail-fast: false
+      matrix:
+        scenario:
+          - 'ember-lts-3.24'
+          - 'ember-lts-3.28'
+          - 'ember-release'
+          - 'ember-beta'
+          - 'ember-canary'
+          - 'ember-classic'
+          # - 'embroider-safe'
+          # - 'embroider-optimized'
+    timeout-minutes: 10
+    steps:
+      - name: Check out a copy of the repo
+        uses: actions/checkout@v3
+
+      - name: Use Node.js ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v3
+        with:
+          cache: 'yarn'
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Test
+        run: yarn test:ember-compatibility ${{ matrix.scenario }} --- yarn test:ember
+
+
+  # try-typescript:
+  #   name: Try TypeScript
+  #   runs-on: ubuntu-latest
+  #   needs: [lint, test-addon]
+  #   strategy:
+  #     fail-fast: false
+  #     matrix:
+  #       scenario:
+  #         - 'typescript-3'
+  #   timeout-minutes: 10
+  #   steps:
+  #     - name: Check out a copy of the repo
+  #       uses: actions/checkout@v3
+
+  #     - name: Use Node.js ${{ env.NODE_VERSION }}
+  #       uses: actions/setup-node@v3
+  #       with:
+  #         cache: 'yarn'
+  #         node-version: ${{ env.NODE_VERSION }}
+
+  #     - name: Install dependencies
+  #       run: yarn install --frozen-lockfile
+
+  #     - name: Test
+  #       run: yarn test:ember-compatibility ${{ matrix.scenario }} --- yarn lint:types --project tsconfig.ts3.json
```

Comment:
@buschtoens I'm not sure if we need this check (i.e. run `yarn lint:types` with a separate TypeScript configuration) and why we had needed it. Could we remove it?